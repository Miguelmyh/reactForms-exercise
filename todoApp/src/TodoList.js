import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

const TodoList = () => {
  const INITIAL_STATE = [];
  const [todos, setTodos] = useState(INITIAL_STATE);

  const addTodo = (newTodo) => {
    setTodos((oldTodo) => [...oldTodo, { ...newTodo, id: uuid() }]);
  };

  const remove = (id) => {
    setTodos(todos.filter((todo) => todo.id !== id));
  };

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      {todos.map((todo) => (
        <Todo key={todo.id} id={todo.id} task={todo.task} remove={remove} />
      ))}
    </div>
  );
};

export default TodoList;
