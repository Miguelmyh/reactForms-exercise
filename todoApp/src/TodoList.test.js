import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

test("should render", () => {
  render(<TodoList />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

test("should add a task", () => {
  const { queryByPlaceholderText, queryByText } = render(<TodoList />);
  const task = queryByPlaceholderText("TodoItem");
  const btn = queryByText("Submit");
  expect(queryByText("clean the kitchen")).not.toBeInTheDocument();
  fireEvent.change(task, { target: { value: "clean the kitchen" } });
  fireEvent.click(btn);
  expect(queryByText("clean the kitchen")).toBeInTheDocument();
});
