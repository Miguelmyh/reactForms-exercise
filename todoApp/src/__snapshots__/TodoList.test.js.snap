// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="todo"
      >
        Todo: 
      </label>
      <input
        id="todo"
        name="task"
        placeholder="TodoItem"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
